File: src/main/resources/filter.functions/filter004.js
filter004 = function(advice) {
    var wait = advice.attributes.get("wait");
    var lifeSpan = advice.attributes.get("timeSpan");
    var plannedExitHour = (advice.getGoodTillTime() % 86400000)/3600000;

    var cred = advice.attributes.get("cred");

    var targetHigh = advice.attributes.get("targetHigh");
    var targetLow = advice.attributes.get("targetLow");
    var targetPnL = targetHigh + targetLow;
    var minAmp = Math.min(targetHigh, targetLow);

    var dP20 = advice.attributes.get("d20");
    var dP50 = advice.attributes.get("d50");

    var priceMovementParameter_1 = (10000 * dP50 * 10000 *dP20);


    if ((20 * wait)/lifeSpan < 7 &&
        wait/60000 < 40 &&
        cred < 60 &&
        lifeSpan/1800000 >= 3 &&
        lifeSpan/1800000 <= 12 &&
        targetPnL <= 0.0016 &&
        minAmp >= -0.0001 &&
        minAmp < 0.0001 &&
        plannedExitHour > 1 &&
        priceMovementParameter_1 >= -1 &&
        priceMovementParameter_1 <= 13){
        return true;
    }

    return false;
}

function getLogarithmicDiscretisation(value){
    value = signum(value) * (Math.log(Math.abs(value) + 1));
    return Math.round(value);
}

function signum(value){
    if(value < 0){
        return -1;
    }
    return 1;
}

File: src/main/java/ai/context/util/trading/version_1/trade/filter/Filter004.java
package ai.context.util.trading.version_1.trade.filter;

import ai.context.util.trading.version_1.MarketMakerPosition;

public class Filter004 implements TradeFilter{
    @Override
    public boolean pass(MarketMakerPosition advice) {

        long wait = (Long)advice.attributes.get("wait");
        double cred = (Double)advice.attributes.get("cred");
        long lifeSpan = (Long)advice.attributes.get("timeSpan");

        long plannedExitHour = (advice.getGoodTillTime() % 3600000)/3600000;
        double targetHigh = (Double)advice.attributes.get("targetHigh");
        double targetLow = (Double)advice.attributes.get("targetLow");
        double targetPnL = targetHigh + targetLow;
        double minAmp = Math.min(targetHigh, targetLow);

        double dP20 = (Double)advice.attributes.get("d20");
        double dP50 = (Double)advice.attributes.get("d50");

        double priceMovementParameter_1 = (int) (10000 * dP50 * 10000 *dP20);

        if ((20 * wait)/lifeSpan < 7 &&
            wait/60000 < 40 &&
            cred < 60 &&
            lifeSpan/1800000 >= 3 &&
            lifeSpan/1800000 <= 12 &&
            targetPnL <= 0.0016 &&
            minAmp >= -0.0001 &&
            minAmp < 0.0001 &&
            plannedExitHour > 1 &&
            priceMovementParameter_1 >= -1 &&
            priceMovementParameter_1 <= 13){
            return true;
        }

        return false;
    }
}
